onnxruntime/contrib_ops/webgpu/quantization/subgroup_matrix_matmul_nbits.cc

```diff
@@ -73,23 +73,82 @@ bool IsSubgroupMatrixConfigSupportedOnIntel(onnxruntime::webgpu::ComputeContext&
   return false;
 }
 
+// This program optimizes the layout of input matrix A(MxK) for SubgroupMatrixLoad, so that all elements of each
+// subgroup matrix(mxk) are arranged continuously in memory.
+// Take "M = 4, K = 4, m = 2, k = 2" as an example, the input matrix A is arranged in row-major order as follows:
+// d00, d01, | d02, d03,
+// d10, d11, | d12, d13,
+// ---------------------
+// d20, d21, | d22, d23,
+// d30, d31, | d32, d33,
+//
+// The layout program rearranges the input matrix A to be in the following order:
+// d00, d01,
+// d10, d11,
+// ---------
+// d02, d03,
+// d12, d13,
+// ---------
+// d20, d21,
+// d30, d31,
+// ---------
+// d22, d23,
+// d32, d33,
+class LayoutProgram final : public Program<LayoutProgram> {
```

Comment:
I think ORT has a different name `Prepack` for the similar thing. using a verb (like prepack or rearrange as mentioned above in the comments) is easier to understand. But this is only my opinion and I am open to different options.

Reply:
`Prepack` sounds good, thanks!